src/csharp.tmLanguage.yml

```diff
@@ -3420,30 +3420,39 @@ repository:
 
   comment:
     patterns:
+    - name: comment.block.documentation.cs
+      begin: (^\s*)(///)(?!/)
+      while: (^\s*)(///)(?!/)
+      captures:
+        1: { name: punctuation.whitespace.comment.leading.cs }
+        2: { name: punctuation.definition.comment.cs }
+      patterns:
+      - include: '#xml-doc-comment'
+    - name: comment.block.documentation.cs
+      begin: (^\s*)(/\*\*)(?!/)
+      end: (^\s*)(\*/)
+      captures:
+        1: { name: punctuation.whitespace.comment.leading.cs }
+        2: { name: punctuation.definition.comment.cs }
+      patterns:
+      - match: (^\s*)(\*)(?!/)
+        captures:
+         1: { name: punctuation.whitespace.comment.leading.cs }
+         2: { name: punctuation.definition.comment.cs }
+      - include: '#xml-doc-comment'
+    - name: comment.line.double-slash.cs
+      begin: (^\s+)?(//).*$
+      beginCaptures:
+        1: { name: punctuation.whitespace.comment.leading.cs }
```

Comment:
Why do we need to highlight leading whitespace?

Reply:
I don't know why. It doesn't seem useful to me. The C# grammar started as a fork of the typescript grammar, and I can see that they emit this scope as well. https://grep.app/search?q=punctuation.whitespace.comment.leading

Reply:
There must be some reason. At least at the moment there's no harm in keeping it.